src/handlers/HeyHandler.ts

```diff
@@ -1,21 +1,35 @@
 import { singleton, inject } from "tsyringe";
-
+import { Either } from "fp-ts/lib/Either";
 import Handler from "./Handler";
 import HandlerConfig from "./HandlerConfig";
 import { GroupMeInfo } from "../services/GroupMeService";
-import { Either } from "fp-ts/lib/Either";
 import MessagingService from "../services/MessagingService";
 
 @singleton()
-export default class HeyHandler implements Handler {
-  private static readonly regexps = [new RegExp("hey loki", "i")];
+export default class HeyHandler extends Handler<GroupMeInfo> {
   readonly config: HandlerConfig = {
-    regexps: HeyHandler.regexps,
+    regexp: new RegExp("hey loki", "i"),
   };
 
-  constructor(@inject("MessagingService") private readonly messagingService: MessagingService) {}
+  constructor(@inject("MessagingService") private readonly messagingService: MessagingService) {
+    super();
+  }
+
+  /**
+   * Returns a boolean saying whether or not the handler should handle the information is is given.
+   * @param groupMeInfo
+   */
+  shouldHandle = (groupMeInfo: GroupMeInfo): boolean => {
+    const passed = this.config.regexp.test(groupMeInfo.text);
+    return passed;
+  };
 
-  async handle(groupMeInfo: GroupMeInfo): Promise<Either<Error, number>> {
-    return this.messagingService.sendMessage(`Howdy ${groupMeInfo.name}!`) as Promise<Either<Error, number>>;
+  /**
+   * Performs its own handler functions as well as calls those it is the root of.
+   * @param groupMeInfo
+   */
+  handle(groupMeInfo: GroupMeInfo): Promise<Either<Error, unknown>>[] {
+    const result = this.messagingService.sendMessage(`Howdy ${groupMeInfo.name}!`);
```

Comment:
I think this needs to awaited
```suggestion
    const result = await this.messagingService.sendMessage(`Howdy ${groupMeInfo.name}!`);
```

Reply:
The await would be redundant. Since you don't resolve in the handle you let the caller resolve the handle which resolves the sendMessage.

Reply:
oh hm yeah guess so